### PR TITLE
NFT Gallery

### PR DESCRIPTION
**Pull request by neuodev**
## Preview 
<details>
<summary> Screenshots </summary>
Screen Shot 2022-07-26 at 2 21 50 PM
Screen Shot 2022-07-26 at 2 22 01 PM
Screen Shot 2022-07-26 at 2 22 07 PM
Screen Shot 2022-07-26 at 2 22 17 PM
n
Screen Shot 2022-07-26 at 2 22 26 PM
</details>

Relevent Task [YOEXT-76](https://emurgo.atlassian.net/browse/YOEXT-76)

### CODE REVIEW

File: packages/yoroi-extension/app/components/wallet/assets/NFTsList.js [hunk 1]
```diff
@@ -109,26 +168,80 @@ function NfTsList({ list, intl }: Props & Intl): Node {
...
 
-function NftCardImage({ image, name }) {
-  let nftImage
-  if (image != null) {
-    const ipfsHash = image.replace('ipfs://', '');
-    nftImage = (<img src={`https://ipfs.io/ipfs/${ipfsHash}`} alt={name} loading="lazy" />);
-  } else {
-    nftImage = (<DefaultNFT />);
-  }
+function isImageExist(imageSrc: string, onload: void => void, onerror: void => void) {
```
Comment (nistadev) [inline packages/yoroi-extension/app/components/wallet/assets/NFTsList.js]: `imageExists` seems like a better way to name this function imo

File: packages/yoroi-extension/app/components/wallet/assets/NFTsList.js [hunk 2]
```diff
@@ -109,26 +168,80 @@ function NfTsList({ list, intl }: Props & Intl): Node {
...
+
+export function NftImage({ imageUrl, name, width, height }: {|
+  imageUrl: string | null,
+  name: string,
+  width: string,
+  height: string
+|}): Node {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  imageUrl = imageUrl !== null ? imageUrl.replace('ipfs://', 'https://ipfs.io/ipfs/'): null;
```
Comment (nistadev) [inline packages/yoroi-extension/app/components/wallet/assets/NFTsList.js]: Reasigning a prop is not a good practice, better create a new variable.

File: packages/yoroi-extension/app/components/wallet/assets/NFTsList.js [hunk 3]
```diff
@@ -109,26 +168,80 @@ function NfTsList({ list, intl }: Props & Intl): Node {
...
+      backgroundColor: 'var(--yoroi-palette-gray-100)',
+      width,
+      height,
+    }}
+    >
+      <DefaultNFT />
+    </Box>
+  )
+
+  if (loading) return <Skeleton variant='rectangular' animation='wave' sx={{ width, height }} />
```
Comment (nistadev) [inline packages/yoroi-extension/app/components/wallet/assets/NFTsList.js:211]: Return the skeleton while loading before the error/imageUrl check.

**Review by Nebyt — APPROVED** [review]
Checked :+1:

**Review by Nebyt — COMMENTED** [review]
/check

File: packages/yoroi-extension/app/components/wallet/assets/NFTsList.js (hunk 3 above)
Comment (neuodev) [reply packages/yoroi-extension/app/components/wallet/assets/NFTsList.js:211]: it is possible that the image NFT might have no URL due to missing metadata. that is why I am returning the `imageUrl` check first,

File: packages/yoroi-extension/app/components/wallet/assets/NFTsList.js (hunk 2 above)
Comment (neuodev) [reply packages/yoroi-extension/app/components/wallet/assets/NFTsList.js]: @nistadev  Good point, Thanks!

**Review by zuzunker — COMMENTED** [review]
/check

**Review by zuzunker — COMMENTED** [review]
/check

**Review by zuzunker — APPROVED** [review]

**Pull request merged**